Allow the location that .so files are searched for for dynamic
loading to be changed via an environment variable. This is to allow
us to load .so's from the host system while building for the target
system.

--- perl-5.8.8/ext/DynaLoader/XSLoader_pm.PL	2007/04/20 09:03:08	1.1
+++ perl-5.8.8/ext/DynaLoader/XSLoader_pm.PL	2007/04/20 09:41:28
@@ -65,6 +65,15 @@
 print OUT <<'EOT';
     my $modpname = join('/',@modparts);
     my $modlibname = (caller())[1];
+    # OE: Allow env to form dynamic loader to look in a different place
+    # This is so it finds the host .so files, not the targets
+    if (defined $ENV{PERLHOSTLIB})
+    {
+        my $hostlib = $ENV{PERLHOSTLIB};
+        print STDERR "*** Module name IN: $modlibname\n";
+        $modlibname =~ s#(?<!/)(\.\./)*lib/#$hostlib#g;
+        print STDERR "*** Module name OUT: $modlibname\n";
+    }
     my $c = @modparts;
     $modlibname =~ s,[\\/][^\\/]+$,, while $c--;	# Q&D basename
     my $file = "$modlibname/auto/$modpname/$modfname.$dl_dlext";
